### ISSUE TITLE
Add some Javadoc to Result.attach() and Result.detach() to explain the semantics

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#3199]
The semantics of `Result.attach()` (and `Result.detach()`) are that all contained `Records` will be attached or detached. This contract should be established in the Javadocs.

---

See also:
http://www.jooq.org/doc/latest/manual/sql-execution/crud-with-updatablerecords/simple-crud/#comment-1349492876
